packages/blockchain-wallet-v4-frontend/src/middleware/analyticsMiddleware.ts

```diff
@@ -195,14 +195,26 @@ const simpleBuyPaymentTypeDictionary = (rawPaymentType: SBPaymentTypes) => {
 
 const getOriginalTimestamp = () => new Date().toISOString()
 
+const id = v4()
+
 const analytics = queuevent<AnalyticsKey, AnalyticsPayload>({
   queueCallback: async (rawEvents) => {
     const res = await fetch('/wallet-options-v4.json')
     const options = await res.json()
 
     const analyticsURL = `${options.domains.api}/events/publish`
 
-    const id = v4()
+    const context: { traits: { nabu_id: string | null } } = {
+      traits: {
+        nabu_id: null
+      }
+    }
+
+    const nabuId = rawEvents.find((event) => event.payload.id)?.payload.id
```

Comment:
if payload is undefined this will throw, can you guarantee it's always there? Also it seems like your `find()` is just looking for the first event with a payload id. Is that what you want it to do?